components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/protocol/nats/NatsInjectHandler.java

```diff
@@ -0,0 +1,151 @@
+import org.apache.synapse.transport.customlogsetter.CustomLogSetter;
+import org.wso2.carbon.context.PrivilegedCarbonContext;
+import org.wso2.carbon.utils.multitenancy.MultitenantConstants;
+
+import java.io.ByteArrayInputStream;
+import java.io.InputStream;
+
+/**
+ * Inject handler to inject NATS message into the sequence.
+ */
+public class NatsInjectHandler {
+    private static final Log log = LogFactory.getLog(NatsInjectHandler.class);
+
+    private String injectingSeq;
+    private String onErrorSeq;
+    private boolean sequential;
+    private SynapseEnvironment synapseEnvironment;
+    private String contentType;
+
+    public NatsInjectHandler(String injectingSeq, String onErrorSeq, boolean sequential,
+            SynapseEnvironment synapseEnvironment, String contentType) {
+        this.injectingSeq = injectingSeq;
+        this.onErrorSeq = onErrorSeq;
+        this.sequential = sequential;
+        this.synapseEnvironment = synapseEnvironment;
+        this.contentType = contentType;
+    }
+
+    /**
+     * Determine the message builder to use, set the message payload to the message context and
+     * inject the message to the sequence
+     */
+    public boolean invoke(Object object, String name, String replyTo, Connection connection) throws SynapseException {
+        try {
+            org.apache.synapse.MessageContext msgCtx = createMessageContext();
+            msgCtx.setProperty(SynapseConstants.INBOUND_ENDPOINT_NAME, name);
+            msgCtx.setProperty(SynapseConstants.IS_INBOUND, true);
+            InboundEndpoint inboundEndpoint = msgCtx.getConfiguration().getInboundEndpoint(name);
+            CustomLogSetter.getInstance().setLogAppender(inboundEndpoint.getArtifactContainerName());
+            MessageContext axis2MsgCtx = ((org.apache.synapse.core.axis2.Axis2MessageContext) msgCtx)
+                    .getAxis2MessageContext();
+
+            // Check if the reply to field is null and send the response the the NATS publisher from the Respond Mediator
+            if (replyTo != null) {
+                msgCtx.setProperty(InboundEndpointConstants.INBOUND_ENDPOINT_RESPONSE_WORKER, new NatsReplySender(replyTo, connection));
+            }
+
+            // Determine the message builder to use
+            Builder builder;
+            if (contentType == null) {
+                printDebugLog("No content type specified. Using SOAP builder.");
+                builder = new SOAPBuilder();
+            } else {
+                int index = contentType.indexOf(';');
+                String type = index > 0 ? contentType.substring(0, index) : contentType;
+                builder = BuilderUtil.getBuilderFromSelector(type, axis2MsgCtx);
+                if (builder == null) {
+                    log.warn("No message builder found for type '" + type + "'. Falling back to SOAP.");
+                    builder = new SOAPBuilder();
+                }
+            }
```

Comment:
Can we move these whole code as a separate util method so that it can be used by other classes aswell

Reply:
Moved to a separate method, Within NATS its used only here.